R/driver.R

```diff
@@ -25,17 +25,27 @@ run_chunk <- function(chunk, all_data) {
 #' @param chunk_data Data produced by chunk
 #' @param chunk_inputs Names of chunk inputs, character
 #' @param promised_outputs Names of chunk's promised outputs, character
-check_chunk_outputs <- function(chunk, chunk_data, chunk_inputs, promised_outputs) {
+#' @param outputs_xml Logical vector: are outputs XML?
+check_chunk_outputs <- function(chunk, chunk_data, chunk_inputs, promised_outputs, outputs_xml) {
   assert_that(is_data_list(chunk_data))
 
   # Check that the chunk has provided required data for all objects
   empty_precursors <- TRUE
   for(obj in names(chunk_data)) {
-    assert_that(tibble::is.tibble(chunk_data[[obj]]))
-    for(at in c(ATTR_TITLE, ATTR_UNITS, ATTR_COMMENTS, ATTR_LEGACY_NAME)) {
-      if(is.null(attr(chunk_data[[obj]], at))) {
-        warning("No '", at, "' attached to ", obj, " - chunk ", chunk)
+    obj_flags <- get_flags(chunk_data[[obj]])
+    # Chunks have to returns tibbles, unless they're tagged as being XML
+    if(!outputs_xml[which(obj == promised_outputs)]) {
```

Comment:
I think @bpbond can answer this one better than I can but I think the short answer is: because we don't trust ourselves and so would like some way of testing you said you are going to give us XML (`outputs_xml`) but did you really do that (has flag `FLAG_XML`).